src/realm/sync/config.hpp

```diff
@@ -167,7 +168,12 @@ struct SyncConfig {
 
     // Will be called after a download message is received and validated by the client but befefore it's been
     // transformed or applied. To be used in testing only.
-    std::function<void(std::weak_ptr<SyncSession>)> on_download_message_received_hook;
+    std::function<void(std::weak_ptr<SyncSession>, const sync::SyncProgress&, int64_t, sync::DownloadBatchState)>
+        on_download_message_received_hook;
+    // Will be called after each bootstrap message is added to the pending bootstrap store, but before_frozen
```

Comment:
this comment is a bit more accurate, but I'm not sure what `before_frozen` has to do with it.